devito/types/separable.py

```diff
@@ -0,0 +1,192 @@
+
+from devito.finite_differences import Differentiable
+from devito.finite_differences.differentiable import Mul, Add
+from devito.tools import as_tuple
+from devito.types.basic import AbstractCachedSymbol
+from devito.types.dense import Function, TimeFunction
+
+__all__ = ['SeparableFunction', 'SeparableTimeFunction']
+
+
+class SeparableFunction(Differentiable, AbstractCachedSymbol):
+    """
+    A Function separable in its Dimensions, ie f(x,y,z) = f(x)*f(y)*f(z)
+
+    Take the same parameters as a Function plus
+
+    Parameters
+    ----------
+    sep: str
```

Comment:
a space is expected between the variable name and the :

Reply:
also, ...sum of FunctionS (plural)

Reply:
how about calling it "op" or "operator" ?

also, probably "separated" should come first

Reply:
>a space is expected between the variable name and the :

wait really, how french... :D

>also, probably "separated" should come first

I don't know, "separated" is the less-likely to be used, general case is don't specify dim only if separable or not and if + or *

>how about calling it "op" or "operator" ?

Can find a better name yes

Reply:
> wait really, how french... :D

just numpydoc convention 😎 

> I don't know, "separated" is the less-likely to be used, general case is don't specify dim only if separable or not and if + or *

oki

> Can find a better name yes

I think ``op`` or ``operator`` is preferable